### DISCUSSION TITLE
Make "Unable to update container" warnings rather than info

### DISCUSSION THREAD

**Discussion by Digital-Grinnell** [containrrr/watchtower#681] [Q&A] [answered]
This is not a bug, but I'm also not sure it's a feature request?  Problem: I'm currently getting lots of notifications of the form...

```
2020-11-16 19:28:12 (info): Unable to update container /xxxxxxxx. Proceeding to next.
2020-11-16 19:28:13 (info): Unable to update container /yyyyyyyy. Proceeding to next.
2020-11-16 19:28:13 (info): Unable to update container /zzzzzzzz. Proceeding to next.
```

I'm not expecting any updates at this time so these messages are just clutter, and there are LOTS of them.  On one platform I've just set my minimum log-level to "warning" and that effectively hides all of this "info" clutter, but when I do update a container I no longer get ANY notification of success or failure now.

So, would it be reasonable to "promote" the "Unable to update container..." messages from info-level to warning-level notifications?

**Comment by piksel** [comment]
I don't see how that solves your problem. Wouldn't that still give you all those "Unable to update container..." notifications, even with minimum log level set to "Warning"?

**Comment by Digital-Grinnell** [comment]
First, thank you for the reply!

You are absolutely correct, this alone would not solve my "clutter" problem, but I feel like it would provide a more accurate indication of the behavior.  It just seems like a "failure" to do something should be differentiated from a "success", if possible, and reporting the failures as warnings or errors, when sufficiently severe, seems consistent with most other packages. 

My real aim here was to open a conversation, and in keeping with that intent... I think what would really be helpful is an additional log-level classification between "info" and "warning", something like "updated", "success", or "confirmed".  That would allow me to ignore "info" and still get confirmation notices when updates are complete.

**Comment by simonaronsson** [comment, marked as answer]
Hey,

Thank you for the report! I definitely agree that we need to look over log levels in general, and this log in particular. We should be able to give more actionable feedback or stay silent in case of false alarms.

If you enable `--debug` and check the logs again, I'm fairly certain you will see rate-limit errors.
We're currently working on fixing that (see #674), and once that's fixed, you shouldn't get even remotely as much of these notifications as you're currently seeing.

Would you mind confirming whether you actually see that being mentioned in the logs if you enable debug?

Thanks,
Simme

**Reply by Digital-Grinnell** [reply]
Yes, I'll apply --debug now, but I think you are correct.  I only see the flurry of "unable to update" messages about once per week, and usually for only an hour or two.  Unfortunately, with refresh set to 5 minutes and 6 containers under watchtower that amounts to as many as 12 x 6 = 72 emails in the span of an hour.  

I will post results here if/when I receive another flurry of messages.  Thanks.

**Comment by Digital-Grinnell** [comment]
OK, I've confirmed that this is a rate-limit issue.  Set my interval accordingly, and am looking forward to #674 implementation.  Thanks all.
